src/main/resources/liquibase/crop_changelog/16_4_1.xml

```diff
@@ -13,6 +13,121 @@
 		</preConditions>
 		<comment>Removes obsolete index germplsm_idx13 created by 16_2_0.xml</comment>
 		<dropIndex tableName="germplsm" indexName="germplsm_idx13"></dropIndex>
+
+	</changeSet>
+
+	<changeSet author="corina" id="v16.4.1-1">
+		<preConditions onFail="MARK_RAN">
+			<sqlCheck expectedResult="1">
+				SELECT
+				COUNT(*)
+				FROM
+				information_schema.routines
+				WHERE
+				routine_schema = DATABASE()
+				AND specific_name = 'populate_entry_type';
+			</sqlCheck>
+		</preConditions>
+		<sql dbms="mysql" splitStatements="false">
+			DROP PROCEDURE populate_entry_type;
+		</sql>
+	</changeSet>
+
+	<changeSet author="corina" id="v16.4.1-2">
+		<preConditions onFail="MARK_RAN">
+			<sqlCheck expectedResult="0">
+				SELECT
+				COUNT(*)
+				FROM
+				information_schema.routines
+				WHERE
+				routine_schema = DATABASE()
+				AND specific_name = 'populate_entry_type';
+			</sqlCheck>
+		</preConditions>
+		<comment>
+			Create SP that adds ENTRY_TYPE variable and value for existing studies without it
+		</comment>
+		<sql dbms="mysql" splitStatements="false">
+			CREATE PROCEDURE populate_entry_type()
+			BEGIN
+			
+				DECLARE proj VARCHAR(11);
+				DECLARE rankno VARCHAR(11);
+				DECLARE loop0_eof BOOLEAN DEFAULT FALSE;
+				DECLARE cur0 CURSOR FOR SELECT p1.project_id,MAX(p2.rank)+1 FROM project p1,projectprop p2 WHERE p1.project_id=p2.project_id AND p1.dataset_type_id=4 AND p1.deleted=0
```

Comment:
More performant to do an INNER JOIN:
`SELECT p1.project_id,MAX(p2.rank)+1 
FROM project p1
INNER JOIN projectprop p2 ON p1.project_id=p2.project_id 
WHERE p1.dataset_type_id=4 AND p1.deleted=0 
				AND NOT EXISTS (SELECT 1 FROM projectprop pp WHERE pp.variable_id=8255 AND pp.project_id=p1.project_id)
				GROUP BY p1.project_id;`